lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+            model.value_left.set_content <+ value_text.map(|t| t.0.clone());
+            value_text_right <- value_text.map(|t| t.1.clone()).gate(&value_text_decimal_visible);
+            model.value_right.set_content <+ value_text_right.map(
+                |t| t.clone().unwrap() // value_text_right is gated by t.1.is_some() in value_text_decimal_visible
+            );
+            value_text_decimal_visibility_change <- value_text_decimal_visible.on_change();
+            eval value_text_decimal_visibility_change (
+                (v) model.set_value_decimal_visible(*v);
+            );
+
+            value_text_left_pos_x <- all2(&model.value_left.width, &model.value_dot.width).map(
+                |(left, dot)| -*left - *dot / 2.0
+            );
+            eval value_text_left_pos_x (
+                (x) model.value_left.set_position_x(*x);
+            );
+            eval model.value_left.height (
+                (h) model.value_left.set_position_y(*h / 2.0);
+            );
+            eval model.value_dot.width (
+                (w) model.value_dot.set_position_x(-*w / 2.0);
+            );
+            eval model.value_dot.height (
+                (h) model.value_dot.set_position_y(*h / 2.0);
+            );
+            eval model.value_dot.width (
+                (w) model.value_right.set_position_x(*w / 2.0);
+            );
+            eval model.value_right.height (
+                (h) model.value_right.set_position_y(*h / 2.0);
+            );
+
+            model.label.set_content <+ input.set_label;
+            eval input.set_label_visible (
+                (v) model.set_label_visible(*v);
+            );
+
+            eval model.label.height (
+                (h) model.label.set_position_y(*h / 2.0);
+            );
+            label_pos_x <- all4(
+                &input.set_width,
+                &input.set_height,
+                &model.label.width,
+                &input.set_label_inside
+            ).map(
+                |(comp_width, comp_height, lab_width, inside)| if *inside {
+                    -comp_width / 2.0 + comp_height / 2.0
+                } else {
+                    -comp_width / 2.0 - comp_height / 2.0 - lab_width
+                }
+            );
+            eval label_pos_x (
+                (x) model.label.set_position_x(*x);
+            );
+
+        }
+
+        self.frp.set_precision(0.1);
+        self.frp.set_precision_step_margin(10.0);
+        self.frp.set_precision_step_size(50.0);
+
+        self.frp.set_value_min(0.0);
+        self.frp.set_value_max(5.0);
+        self.frp.set_value_default(0.5);
+        self.frp.set_value(0.5);
+
+        self.frp.set_background_color(color::Lcha(0.8, 0.0, 0.0, 1.0));
+        self.frp.set_slider_color(color::Lcha(0.5, 0.5, 0.0, 1.0));
+        self.frp.set_slider_enabled(true);
+
+        self.frp.set_label_visible(true);
+        self
+    }
+}
+
+
+
+// ========================
+// === Helper functions ===
```

Comment:
We are banning sections names "helpers" and "utils", as they tend to grow and contain a lot of stuff that is not nicely structured. Everything can be "helper" or "util". Try finding better names for sections, please.